jdbc/src/test/java/com/interface21/jdbc/core/JdbcTemplateTest.java

```diff
@@ -130,16 +130,16 @@ void update() throws SQLException {
         String sql = "UPDATE test_user SET name = ? WHERE id = ?";
         when(pstmt.executeUpdate()).thenReturn(1);
 
-        jdbcTemplate.update(sql, "aru", 1);
+        int updatedRows = jdbcTemplate.update(sql, "aru", 1);
 
         assertAll(
+                () -> assertThat(updatedRows).isEqualTo(1),
```

Comment:
이건 레벨 1에서 받았던 질문인데요, 생각해볼 만 한 내용이라 남겨둡니다. Mock를 사용하게 되면 테스트하는 것에 대한 경계가 상당히 모호해질 수 있습니다. 어떻게 보면 덧셈 연산을 검증할 때, `1+1=2`를 모킹하고 그 값이 2인지를 확인하는 방식이죠. 테스트는 두꺼워질 수 있지만, 정말 이 테스트가 의미있는가? 라고 생각하면 그렇지 않은 경우도 있습니다. 테스트하고자하는 것이 무엇인지를 확실하게 잡고가면 도움이 되겠네요.

위와 같은 바람직하지 못한 검증을 Tautological Test라고도 하는데요, 한 번 찾아보세요! 코드에 Mock을 해둔 뒤, Mock을 검증하고있지 않은가 생각해보는 것도 좋겠습니다 👍🏻

Reply:
오 아루가 알려준 `Tautological Test`에 대해 알아봤는데
In this case, it would be better to do the calculation in our head and write the answer into the test
가 인상깊네요

모킹한 값에 대한 테스트를 하는 것은  무조건 성공하는 테스트를 테스트하는 것이나 다름없이니 불필요한 테스트겠네요
좋은 키워드 감사해요 👍